bindings/c/test/unit/unit_tests.cpp

```diff
@@ -0,0 +1,1575 @@
+/*
+ * unit_tests.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unit tests for the FoundationDB C API.
+
+#define FDB_API_VERSION 620
+#include <foundationdb/fdb_c.h>
+#include <assert.h>
+#include <string.h>
+
+#include <condition_variable>
+#include <iostream>
+#include <map>
+#include <mutex>
+#include <optional>
+#include <stdexcept>
+#include <thread>
+#include <tuple>
+#include <vector>
+
+#define DOCTEST_CONFIG_IMPLEMENT
+#define DOCTEST_CONFIG_NO_EXCEPTIONS_BUT_WITH_ALL_ASSERTS
```

Comment:
I think the right default here might be to not define DOCTEST_CONFIG_NO_EXCEPTIONS_BUT_WITH_ALL_ASSERTS and DOCTEST_CONFIG_NO_POSIX_SIGNALS. The reason for defining these is to stop doctest from intercepting fatal signals like segfaults e.g., so that running the binary under a debugger is more useful. I think the doctest signal handlers might be better for running in CI, since otherwise it can't tell you which test caused the segfault.